### PR TITLE
👷 Configure a Travis CI build

### LINKED ISSUE #35: Hook up Travis CI to run tests on pull requests
Once we have the `npm test` command working (see #33) we can hook up [Travis CI](https://travis-ci.org/) so tests are run on pull requests. If we add a ".travis.yaml" file to the root of the project with the following contents, it should work.

~~~yaml
language: node_js
sudo: false
script:
  - npm test
~~~

We may have to add a ".nvmrc" file to the root of the project with the following contents to tell Travis CI what version of Node to use.

~~~text
4.4.0
~~~

### LINKED ISSUE #36: Hook up Travis CI to lint code on pull requests
Once we have the `npm test` command working in Travis CI (see #35) we can add lint coverage by extending the "scripts" section of the ".travis.yaml" file with a `npm run lint` command. The linter should run before the tests are run.

### LINKED ISSUE #37: Update README to include build status icon
Once Travis CI is wired up to run on pull requests (see #35) we can update the README file to include a build status icon. The following Markdown snippet should go at the top of the README right after the "Awsaml" header.

~~~text
[![Build Status][build-status-image]][build-status-url]
~~~

The `build-status-image` and `build-status-url` links should be defined at the bottom of the README.

~~~text
[build-status-image]: https://travis-ci.org/rapid7/awsaml.svg?branch=master
[build-status-url]: https://travis-ci.org/rapid7/awsaml
~~~

### PR DESCRIPTION
**Pull request by erran-r7**
* Fix #35.
* Fix #36.
* Fix #37.

### CODE REVIEW

**Comment by fmitchell-r7** [comment]
@erran, it looks like `npm install` failed on Travis CI when compiling native modules. We might have to [set up compiler requirements](https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#Node.js-v4-(or-io.js-v3)-compiler-requirements) in the `.travis.yaml` file.

**Comment by erran-r7** [comment]
> @erran, it looks like npm install failed on Travis CI when compiling native modules. We might have to set up compiler requirements in the .travis.yaml file.
> View changes

@fmitchell-r7 you're right about that! 👍 Looks like I'll need to install `istanbul` as a dependency. (I'll run the script locally before pushing this time 😅)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+language: node_js
+env:
+  - CXX=g++-4.8
+sudo: false
+script:
+  - npm test
+  - npm run lint
+  - npm run cover
+after_success:
+  npm run report
```
Comment (erran-r7) [inline .travis.yml]: This command fails currently due to the repository not being configured in `coveralls.io`. 😫

Comment (cdoughty-r7) [inline .travis.yml]: coveralls enabled

**Comment by fmitchell-r7** [comment]
@erran, thanks for putting this together! 👍 :ship: it.

@cdoughty-r7, thanks for getting coveralls.io enabled!

**Comment by erran-r7** [comment]
> @cdoughty-r7, thanks for getting coveralls.io enabled!

Yep, cheers Chris! 😄 🎉

**Pull request merged**